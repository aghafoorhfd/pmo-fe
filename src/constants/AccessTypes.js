export const ACCESS_TYPES = {
  SUPER_ADMIN: 'SUPER_ADMIN',
  ADMIN: 'ADMIN',
  PROJECT_MANAGER: 'PROJECT_MANAGER',
  RESOURCE_MANAGER: 'RESOURCE_MANAGER',
  CONFLICT_MANAGER: 'CONFLICT_MANAGER',
  GENERAL_USER: 'GENERAL_USER'
};
