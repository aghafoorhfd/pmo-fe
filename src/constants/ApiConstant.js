import { env } from '../configs/EnvironmentConfig';

export const REACT_APP_BASE_URL = env.REACT_APP_URL;

export const USER_SERVICE_BASE_URL = env.USER_SERVICE;
export const USER_SERVICE = `${USER_SERVICE_BASE_URL}/user_service/api/users`;
export const USER_PROFILE = `${USER_SERVICE}/user-profile`;
export const REGISTER_USER = `${USER_SERVICE}/register`;
export const FORGOT_PASSWORD = `${USER_SERVICE}/forgot-password`;
export const VERIFY_EMAIL = `${USER_SERVICE}/verify-email`;
export const SET_CREDENTIALS = `${USER_SERVICE}/credentials`;
export const LOGIN = `${USER_SERVICE}/login`;

export const AUTH_SERVICE_BASE_URL = env.AUTH_SERVICE;
export const AUTH_SERVICE = `${AUTH_SERVICE_BASE_URL}/auth_service/api/users`;
export const RESET_CREDENTIALS = `${AUTH_SERVICE}/reset-password`;

export const GLOBAL_SERVICE_BASE_URL = env.GLOBAL_SERVICE;
export const GLOBAL_SERVICE = `${GLOBAL_SERVICE_BASE_URL}/global_service/api`;

export const PROJECT_SERVICE_BASE_URL = env.PROJECT_SERVICE;
export const PROJECT_SERVICE = `${PROJECT_SERVICE_BASE_URL}/project_service/api`;
export const ADD_PROJECT = `${PROJECT_SERVICE}/projects`;
export const INVITE_STAKEHOLDERS = `${PROJECT_SERVICE}/projects/invite-stakeholders`;
export const PROJECT_RESOURCES = `${PROJECT_SERVICE}/projects/resources-stats`;
export const PROJECT_CONFLICTS = `${PROJECT_SERVICE}/projects/conflicts-stats`;
export const COMPANY_METHODOLOGY = `${PROJECT_SERVICE}/company/methodology`;

export const PAYMENT_SERVICE_BASE_URL = env.PAYMENT_SERVICE;
export const PAYMENT_SERVICE = `${PAYMENT_SERVICE_BASE_URL}/paymentgateway_service/api`;

export const SUBSCRIPTION_SERVICE_BASE_URL = env.SUBSCRIPTION_SERVICE;
export const SUBSCRIPTION_SERVICE = `${SUBSCRIPTION_SERVICE_BASE_URL}/subscription_service/api`;

export const RESOURCE_SERVICE_BASE_URL = env.RESOURCE_SERVICE_API_ENDPOINT_URL;
export const RESOURCE_SERVICE = `${RESOURCE_SERVICE_BASE_URL}/resource_service/api`;
export const RESOURCE_TEAM = `${RESOURCE_SERVICE}/resource-teams`;
export const RESOURCE_REQUEST = `${RESOURCE_SERVICE}/project-manager/resources`;

export const PAYMENT_CONFIRMATION_URL = '/app/dashboards/payment-confirmation';
