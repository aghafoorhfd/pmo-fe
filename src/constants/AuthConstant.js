export const AUTH_TOKEN = 'auth_token';
export const LOGIN = '/auth/login';
export const SIGN_UP = '/auth/register';
