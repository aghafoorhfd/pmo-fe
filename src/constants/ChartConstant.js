export const COLOR_1 = '#3e82f7'; // blue
export const COLOR_2 = '#04d182'; // cyan
export const COLOR_3 = '#ff6b72'; // volcano
export const COLOR_4 = '#ffc107'; // gold
export const COLOR_5 = '#a461d8'; // purple
export const COLOR_6 = '#fa8c16'; // orange
export const COLOR_7 = '#17bcff'; // geekblue
export const COLOR_8 = '#A5A5A5'; // grey
export const COLOR_9 = '#FFEF5E'; // orange
export const COLOR_10 = '#509CF5'; // light-blue
export const COLOR_11 = '#D0D5DD'; // light Grey
export const COLOR_12 = '#039855'; // green
export const COLOR_13 = '#D92D20'; // red

export const COLOR_1_LIGHT = 'rgba(62, 130, 247, 0.15)';
export const COLOR_2_LIGHT = 'rgba(4, 209, 130, 0.1)';
export const COLOR_3_LIGHT = 'rgba(222, 68, 54, 0.1)';
export const COLOR_4_LIGHT = 'rgba(255, 193, 7, 0.1)';
export const COLOR_5_LIGHT = 'rgba(139, 75, 157, 0.1)';
export const COLOR_6_LIGHT = 'rgba(250, 140, 22, .1)';
export const COLOR_7_LIGHT = 'rgba(23, 188, 255, 0.15)';

export const COLORS = [
  COLOR_1,
  COLOR_2,
  COLOR_3,
  COLOR_4,
  COLOR_5,
  COLOR_6,
  COLOR_7,
  COLOR_8,
  COLOR_9,
  COLOR_10,
  COLOR_11,
  COLOR_12,
  COLOR_13
];

export const COLORS_LIGHT = [
  COLOR_1_LIGHT,
  COLOR_2_LIGHT,
  COLOR_3_LIGHT,
  COLOR_4_LIGHT,
  COLOR_5_LIGHT,
  COLOR_6_LIGHT,
  COLOR_7_LIGHT
];

export const COLOR_AXES = '#edf2f9';
export const COLOR_TEXT = '#455560';

export const apexLineChartDefaultOption = {
  chart: {
    zoom: {
      enabled: false
    },
    toolbar: {
      show: false
    }
  },
  colors: [...COLORS],
  dataLabels: {
    enabled: false
  },
  stroke: {
    width: 3,
    curve: 'smooth',
    lineCap: 'round'
  },
  legend: {
    position: 'top',
    horizontalAlign: 'right',
    offsetY: -15,
    itemMargin: {
      vertical: 20
    },
    tooltipHoverFormatter(val, opts) {
      return `${val} - ${opts.w.globals.series[opts.seriesIndex][opts.dataPointIndex]}`;
    }
  },
  xaxis: {
    categories: []
  },
  grid: {
    xaxis: {
      lines: {
        show: true
      }
    },
    yaxis: {
      lines: {
        show: false
      }
    }
  }
};

export const apexAreaChartDefaultOption = { ...apexLineChartDefaultOption };

export const apexBarChartDefaultOption = {
  chart: {
    zoom: {
      enabled: false
    },
    toolbar: {
      show: false
    }
  },
  plotOptions: {
    bar: {
      horizontal: false,
      columnWidth: '25px',
      startingShapre: 'rounded',
      endingShape: 'rounded'
    }
  },
  colors: [...COLORS],
  dataLabels: {
    enabled: false
  },
  stroke: {
    show: true,
    width: 6,
    curve: 'smooth',
    colors: ['transparent']
  },
  legend: {
    position: 'top',
    horizontalAlign: 'right',
    offsetY: -15,
    inverseOrder: true,
    itemMargin: {
      vertical: 20
    },
    tooltipHoverFormatter(val, opts) {
      return `${val} - ${opts.w.globals.series[opts.seriesIndex][opts.dataPointIndex]}`;
    }
  },
  xaxis: {
    categories: []
  },
  fill: {
    opacity: 1
  },
  tooltip: {
    y: {
      formatter: (val) => `${val}`
    }
  }
};

export const apexPieChartDefaultOption = {
  colors: [...COLORS],
  plotOptions: {
    pie: {
      donut: {
        size: '60%',
        labels: {
          show: true
        }
      }
    }
  },
  labels: [],
  dataLabels: {
    enabled: true,
    formatter: (val) => `${val.toFixed(0)}%`
  },
  legend: {
    show: true,
    onItemHover: {
      highlightDataSeries: false
    }
  },
  chart: {
    fontFamily: 'Inter',
    foreColor: 'rgb(69, 85, 96)'
  },
  stroke: {
    width: 0
  }
};

export const apexSparklineChartDefultOption = {
  chart: {
    type: 'line',
    sparkline: {
      enabled: true
    }
  },
  stroke: {
    width: 2,
    curve: 'smooth'
  },
  tooltip: {
    fixed: {
      enabled: false
    },
    x: {
      show: false
    },
    y: {
      title: {
        formatter() {
          return '';
        }
      }
    },
    marker: {
      show: false
    }
  }
};
