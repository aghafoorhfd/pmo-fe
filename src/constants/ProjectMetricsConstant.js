export const PROJECT_STATES = 'Project States';
export const PROJECT_STATUS = 'Project Status';
export const PROJECT_OUTLOOK = 'Project Outlook';

export const tagColors = {
  [PROJECT_STATES]: ['#989898', '#4674ca', '#b0d2a1', '#989898'],
  [PROJECT_STATUS]: ['#b0d2a1', '#FFCCCC']
};

export const initialCompanyConfiguration = {
  projectStages: { selected: [], notSelected: [] },
  sponsoringDepartment: { selected: [], notSelected: [] },
  milestones: { selected: [], notSelected: [] },
  resourceDiscipline: { selected: [], notSelected: [] },
  projectCategory: { selected: [], notSelected: [] },
  projectPriorityLevel: { selected: [], notSelected: [] }
};

export const initialGlobalConfiguration = {
  projectStages: [],
  sponsoringDepartment: [],
  milestones: [],
  projectStatus: [],
  projectStates: [],
  projectCategory: [],
  projectPriorityLevel: []
};

export const methodologyType = {
  AGILE: 'agile',
  SDLC: 'sdlc',
  AGILE_CAPS: 'AGILE',
  SDLC_CAPS: 'SDLC',
  AGILE_SDLC: 'AGILE_SDLC'
};
