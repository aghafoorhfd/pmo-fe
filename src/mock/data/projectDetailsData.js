export const ProjectDetails = {
  companyId: 'd153c439-2bc8-4ede-a092-26955c9d443b',
  projectName: 'newPROJ',
  overview: 'sample overview',
  projectNumber: 'A13de3',
  category: 'samplecategory',
  priority: 'High',
  department: 'Software',
  theme: 'Simplify Management',
  currentStatus: 'Active',
  projectCadence: 'Agile',
  outlook: 'orange',
  outlookReason: 'resource shortage',
  currentState: 'Confirmed',
  currentStage: 'Planning',
  projectManager: 'adeeb',
  techLead: {
    id: '3368898a-f144-421a-8e53-ae5d0c6f84ca',
    firstName: 'adeeb',
    lastName: 'khalid',
    email: 'abc@nisum.com',
    phoneNumber: '1213',
    accessType: 'ADMIN',
    companyId: '1fdf3e68-3592-4252-afec-069ef1bda957'
  },
  productManager: 'adeeb',
  stakeHolders: [{
    id: '3368898a-f144-421a-8e53-ae5d0c6f84ca',
    firstName: 'adeeb',
    lastName: 'khalid',
    email: 'abc@nisum.com',
    phoneNumber: '1213',
    accessType: 'ADMIN',
    companyId: '1fdf3e68-3592-4252-afec-069ef1bda957'
  }]
};
export const ProjectDetailsStore = {
  user: {
    message: '',
    showMessage: false,
    status: 'success'
  },
  auth: {
    userProfile: {
      id: '8f265fcb-42b2-4499-a087-2aa17dc6cf71',
      companyId: 'bc4ba09f-7fde-43c1-979f-8616e1ae0a84',
      accessType: 'SUPER_ADMIN'
    }
  },
  projectDetails: {
    loading: false,
    generalUsers: [{
      id: '3368898a-f144-421a-8e53-ae5d0c6f84ca',
      firstName: 'Hafiz',
      lastName: 'Zeeshan',
      email: 'ddds@yopmail.com'
    }],
    privileges: {},
    message: '',
    showMessage: false,
    status: 'success'
  }
};
