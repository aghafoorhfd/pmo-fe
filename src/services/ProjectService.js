import fetch from 'auth/FetchInterceptor';
import {
  PROJECT_SERVICE, ADD_PROJECT, INVITE_STAKEHOLDERS,
  PROJECT_RESOURCES, PROJECT_CONFLICTS, COMPANY_METHODOLOGY
} from 'constants/ApiConstant';
import { initialPaginationConfiguration, PROJECT_METRICS } from 'constants/MiscConstant';

const { page: initialPage, pageSize: initialPageSize } = initialPaginationConfiguration;

const ProjectService = {

  getProjectDetails(projectId) {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/${projectId}`,
      method: 'get'
    });
  },

  addProject(data) {
    return fetch({
      url: ADD_PROJECT,
      method: 'post',
      data
    });
  },

  updateProject({ projectId, data }) {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/${projectId}`,
      method: 'put',
      data
    });
  },

  inviteStakeHolders(data) {
    return fetch({
      url: INVITE_STAKEHOLDERS,
      method: 'post',
      data
    });
  },

  getProjectResources() {
    return fetch({
      url: PROJECT_RESOURCES,
      method: 'get'
    });
  },

  getProjectConflicts() {
    return fetch({
      url: PROJECT_CONFLICTS,
      method: 'get'
    });
  },

  getProjectList({
    page = initialPage, pageSize = initialPageSize, filterAnd = '', excludeProjectManagerIds = null, filterOr = ''
  }) {
    const filters = [filterOr ? `filterOr=${filterOr}` : '', filterAnd ? `filterAnd=${filterAnd}` : '', excludeProjectManagerIds ? `excludeProjectManagerIds=${excludeProjectManagerIds}` : ''].filter((query) => query);

    return fetch({
      url: `${PROJECT_SERVICE}/projects/${page}/${pageSize}?${filters.join('&')}`,
      method: 'get'
    });
  },

  getProjectTimeLine(projectId) {
    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/metrics`,
      method: 'get'
    });
  },

  getProjectRanges(range) {
    return fetch({
      url: `${COMPANY_METHODOLOGY}/sprints?${range}`,
      method: 'get'
    });
  },

  getProjectQuarters(range) {
    return fetch({
      url: `${COMPANY_METHODOLOGY}/quarters?${range}`,
      method: 'get'
    });
  },

  getRisks(riskStatus, pageNumber, pageSize) {
    return fetch({
      url: `${PROJECT_SERVICE}/risks/${pageNumber}/${pageSize}`,
      method: 'get',
      params: { riskStatus }
    });
  },

  getAllRisks(selectedTab, pageNumber, pageSize, searchTerm) {
    let filterAndParam = '';
    if (searchTerm) {
      const searchTermUri = encodeURIComponent(`conflictName|${searchTerm}`);
      filterAndParam = `&filterAnd=${searchTermUri}`;
    }
    const filterOrParam = encodeURIComponent(`conflictStatus|${selectedTab}`);
    return fetch({
      url: `${PROJECT_SERVICE}/risks/search/${pageNumber}/${pageSize}?filterOr=${filterOrParam}${filterAndParam}`,
      method: 'get'
    });
  },

  getMyRisks(
    pageNumber,
    pageSize,
    selectedTab,
    filterOr = ''
  ) {
    const filterAndParam = encodeURIComponent(`conflictStatus|${selectedTab}`);
    return fetch({
      url: `${PROJECT_SERVICE}/risks/search/${pageNumber}/${pageSize}?filterAnd=${filterAndParam}&filterOr=${filterOr}`,
      method: 'get'
    });
  },

  getGraphStats(conflictStatus, allConflicts = false) {
    return fetch({
      url: `${PROJECT_SERVICE}/risks/stats?conflictStatType=${conflictStatus}&allConflicts=${allConflicts}`,
      method: 'get'
    });
  },

  addNewRisk(data) {
    return fetch({
      url: `${PROJECT_SERVICE}/risks`,
      method: 'post',
      data
    });
  },

  updateRiskDetails(data) {
    return fetch({
      url: `${PROJECT_SERVICE}/risks/${data?.conflictId}`,
      method: 'put',
      data
    });
  },

  getProjectNames() {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/names`,
      method: 'get'
    });
  },

  getRiskDetails(conflictId) {
    return fetch({
      url: `${PROJECT_SERVICE}/risks/${conflictId}`,
      method: 'get'
    });
  },

  getRiskHistoryDetails(conflictId) {
    return fetch({
      url: `${PROJECT_SERVICE}/audit/risks/${conflictId}/history`,
      method: 'get'
    });
  },

  getProjectHistoryDetail({ projectId, page, pageSize }) {
    return fetch({
      url: `${PROJECT_SERVICE}/audit/projects/${projectId}/history/${page}/${pageSize}`,
      method: 'get'
    });
  },

  getOtherImpactedProjects() {
    return fetch({
      url: `${PROJECT_SERVICE}/projects`,
      method: 'get'
    });
  },

  addStages(projectId, body) {
    const data = { metricType: PROJECT_METRICS.STAGES, ...body };
    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/metrics`,
      method: 'post',
      data
    });
  },

  getStageDetails(projectId, stageName) {
    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/metrics/${stageName}`,
      method: 'get'
    });
  },

  updateStage(projectId, body) {
    const data = { ...body };

    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/metrics`,
      method: 'put',
      data
    });
  },
  updateSubStage(projectId, body) {
    const data = { ...body };

    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/sub-stages/dependency`,
      method: 'put',
      data
    });
  },

  addMilestones(projectId, body) {
    const data = { metricType: PROJECT_METRICS.MILESTONES, ...body };
    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/metrics`,
      method: 'post',
      data
    });
  },

  updateMilestone(projectId, data) {
    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/metrics`,
      method: 'put',
      data
    });
  },

  deleteMetric(projectId, metricType, metricName) {
    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/metrics/${metricType}/${metricName}`,
      method: 'delete'
    });
  },

  addDependency(projectId, data) {
    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/stages/dependency`,
      method: 'put',
      data
    });
  },

  getProjectAgileDetails() {
    return fetch({
      url: COMPANY_METHODOLOGY,
      method: 'get'
    });
  },

  saveProjectAgileDetails(data) {
    return fetch({
      url: COMPANY_METHODOLOGY,
      method: 'post',
      data
    });
  },

  updateProjectAgileDetails(data) {
    return fetch({
      url: COMPANY_METHODOLOGY,
      method: 'put',
      data
    });
  },

  getProjectConfigurations() {
    return fetch({
      url: `${PROJECT_SERVICE}/company/metrics`,
      method: 'get'
    });
  },

  updateProjectMetrics(data, method, metric = 'PROJECT_STAGES') {
    const metricParam = `/${metric}`;
    const metricQueryParam = `?metric=${metric}`;
    const url = `${PROJECT_SERVICE}/company/metrics${(method === 'put') ? metricParam : metricQueryParam}`;
    return fetch({
      url,
      method,
      data
    });
  },
  jiraLinkAccount(projectId) {
    return fetch({
      url: `${PROJECT_SERVICE}/jira/${projectId}/link-account`,
      method: 'get'
    });
  },

  jiraCallback(data) {
    return fetch({
      url: `${PROJECT_SERVICE}/jira/callback`,
      method: 'post',
      data
    });
  },
  getUserPendingWorkStatistics(projectId, projectKey) {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/${projectId}/jira/users/pending-work-stats?projectKey=${projectKey}`,
      method: 'get'
    });
  },
  addJiraItem(projectId, data) {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/${projectId}/jira-sync/work-item`,
      method: 'put',
      data
    });
  },
  getMyWorkAllocation() {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/work-allocations`,
      method: 'get'
    });
  },
  syncJira(projectId, projectKey) {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/${projectId}/jira-sync/statistics?projectKey=${projectKey}`,
      method: 'put'
    });
  },

  getJiraIssues(projectId, projectKey) {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/${projectId}/jira/issues?projectKey=${projectKey}`,
      method: 'get'
    });
  },

  getProjectAssignedResources(projectId) {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/${projectId}/stage-resources`,
      method: 'get'
    });
  },

  getStatsForSprintAndProjects() {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/work-allocations`
    });
  },

  getProjectNo() {
    return fetch({
      url: `${PROJECT_SERVICE}/projects/project-number`,
      method: 'get'
    });
  },
  removeMetricDependency(projectId, data) {
    return fetch({
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/metrics/dependency`,
      method: 'delete',
      data
    });
  }
};
export default ProjectService;
