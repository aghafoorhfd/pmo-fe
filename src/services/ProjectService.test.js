import fetch from 'auth/FetchInterceptor';
import ProjectService from 'services/ProjectService';
import {
  PROJECT_SERVICE,
  ADD_PROJECT,
  INVITE_STAKEHOLDERS,
  COMPANY_METHODOLOGY
} from 'constants/ApiConstant';
import { ProjectDetails } from 'mock/data/projectDetailsData';
import { methodologyType } from 'constants/ProjectMetricsConstant';

jest.mock('auth/FetchInterceptor');
jest.mock('utils/utils');

describe('Auth Service', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const metric = 'PROJECT_STAGES';
  const agileData = { methodologyType: methodologyType.AGILE_SDLC };
  const metricsData = { metric: ['stages'] };

  it('should call get Project details handler ', () => {
    const projectId = 'ProjectID';
    ProjectService.getProjectDetails(projectId);

    expect(fetch).toBeCalledWith({
      method: 'get',
      url: `${PROJECT_SERVICE}/projects/${projectId}`
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call add Project details handler ', () => {
    ProjectService.addProject(ProjectDetails);

    expect(fetch).toBeCalledWith({
      method: 'post',
      url: ADD_PROJECT,
      data: ProjectDetails
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });
  it('should call update Project details handler ', () => {
    const projectId = '123';
    ProjectService.updateProject({
      projectId,
      data: ProjectDetails
    });

    expect(fetch).toBeCalledWith({
      method: 'put',
      url: `${PROJECT_SERVICE}/projects/${projectId}`,
      data: ProjectDetails
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });
  it('should call invite stake holders handler ', () => {
    const projectName = 'abc';
    ProjectService.inviteStakeHolders({
      projectName,
      stakeHolders: ProjectDetails.stakeHolders
    });

    expect(fetch).toBeCalledWith({
      method: 'post',
      url: INVITE_STAKEHOLDERS,
      data: {
        projectName,
        stakeHolders: ProjectDetails.stakeHolders
      }
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call get Project Timeline handler ', () => {
    const projectId = 'ProjectID';
    ProjectService.getProjectTimeLine(projectId);

    expect(fetch).toBeCalledWith({
      method: 'get',
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/metrics`
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call get Project ranges handler ', () => {
    const projectId = 'ProjectID';
    ProjectService.getProjectRanges(projectId);

    expect(fetch).toBeCalledWith({
      method: 'get',
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/sprint`
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call get Project Quarter Ranges', () => {
    const projectId = 'ProjectID';
    ProjectService.getProjectQuarters(projectId);

    expect(fetch).toBeCalledWith({
      method: 'get',
      url: `${PROJECT_SERVICE}/timelines/projects/${projectId}/`
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call getProjectAgileDetails with the get method', () => {
    ProjectService.getProjectAgileDetails();
    expect(fetch).toBeCalledWith({
      method: 'get',
      url: COMPANY_METHODOLOGY
    });
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });
  it('should call saveProjectAgileDetails with the post method', () => {
    ProjectService.saveProjectAgileDetails(agileData);
    expect(fetch).toBeCalledWith({
      method: 'post',
      url: COMPANY_METHODOLOGY,
      data: agileData
    });
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });
  it('should call updateProjectAgileDetails with the put method', () => {
    ProjectService.updateProjectAgileDetails(agileData);
    expect(fetch).toBeCalledWith({
      method: 'put',
      url: COMPANY_METHODOLOGY,
      data: agileData
    });
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });
  it('should call getProjectConfigurations endpoint ', () => {
    ProjectService.getProjectConfigurations();
    expect(fetch).toBeCalledWith({
      method: 'get',
      url: `${PROJECT_SERVICE}/company/metrics`
    });
    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });

  it('should call updateProjectMetrics endpoint with put http method', () => {
    ProjectService.updateProjectMetrics(metricsData, 'put', metric);

    expect(fetch).toBeCalledWith({
      method: 'put',
      url: `${PROJECT_SERVICE}/company/metrics/${metric}`,
      data: metricsData
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });
  it('should call updateProjectMetrics endpoint with Post http method', () => {
    ProjectService.updateProjectMetrics(metricsData, 'post', metric);

    expect(fetch).toBeCalledWith({
      method: 'post',
      url: `${PROJECT_SERVICE}/company/metrics?metric=${metric}`,
      data: metricsData
    });

    expect(fetch).toHaveBeenCalledTimes(1);
    expect(fetch).toMatchSnapshot();
  });
});
