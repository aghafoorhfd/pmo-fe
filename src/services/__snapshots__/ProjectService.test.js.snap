// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Service should call add Project details handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "category": "samplecategory",
          "companyId": "d153c439-2bc8-4ede-a092-26955c9d443b",
          "currentStage": "Planning",
          "currentState": "Confirmed",
          "currentStatus": "Active",
          "department": "Software",
          "outlook": "orange",
          "outlookReason": "resource shortage",
          "overview": "sample overview",
          "priority": "High",
          "productManager": "adeeb",
          "projectCadence": "Agile",
          "projectManager": "adeeb",
          "projectName": "newPROJ",
          "projectNumber": "A13de3",
          "stakeHolders": [
            {
              "accessType": "ADMIN",
              "companyId": "1fdf3e68-3592-4252-afec-069ef1bda957",
              "email": "abc@nisum.com",
              "firstName": "adeeb",
              "id": "3368898a-f144-421a-8e53-ae5d0c6f84ca",
              "lastName": "khalid",
              "phoneNumber": "1213",
            },
          ],
          "techLead": {
            "accessType": "ADMIN",
            "companyId": "1fdf3e68-3592-4252-afec-069ef1bda957",
            "email": "abc@nisum.com",
            "firstName": "adeeb",
            "id": "3368898a-f144-421a-8e53-ae5d0c6f84ca",
            "lastName": "khalid",
            "phoneNumber": "1213",
          },
          "theme": "Simplify Management",
        },
        "method": "post",
        "url": "undefined/project_service/api/projects",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call get Project Timeline handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "undefined/project_service/api/timelines/projects/ProjectID/metrics",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call get Project details handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "undefined/project_service/api/projects/ProjectID",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call get Project ranges handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "method": "get",
        "url": "undefined/project_service/api/timelines/projects/ProjectID/sprint",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call invite stake holders handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "projectName": "abc",
          "stakeHolders": [
            {
              "accessType": "ADMIN",
              "companyId": "1fdf3e68-3592-4252-afec-069ef1bda957",
              "email": "abc@nisum.com",
              "firstName": "adeeb",
              "id": "3368898a-f144-421a-8e53-ae5d0c6f84ca",
              "lastName": "khalid",
              "phoneNumber": "1213",
            },
          ],
        },
        "method": "post",
        "url": "undefined/project_service/api/projects/invite-stakeholders",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Auth Service should call update Project details handler  1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "data": {
          "category": "samplecategory",
          "companyId": "d153c439-2bc8-4ede-a092-26955c9d443b",
          "currentStage": "Planning",
          "currentState": "Confirmed",
          "currentStatus": "Active",
          "department": "Software",
          "outlook": "orange",
          "outlookReason": "resource shortage",
          "overview": "sample overview",
          "priority": "High",
          "productManager": "adeeb",
          "projectCadence": "Agile",
          "projectManager": "adeeb",
          "projectName": "newPROJ",
          "projectNumber": "A13de3",
          "stakeHolders": [
            {
              "accessType": "ADMIN",
              "companyId": "1fdf3e68-3592-4252-afec-069ef1bda957",
              "email": "abc@nisum.com",
              "firstName": "adeeb",
              "id": "3368898a-f144-421a-8e53-ae5d0c6f84ca",
              "lastName": "khalid",
              "phoneNumber": "1213",
            },
          ],
          "techLead": {
            "accessType": "ADMIN",
            "companyId": "1fdf3e68-3592-4252-afec-069ef1bda957",
            "email": "abc@nisum.com",
            "firstName": "adeeb",
            "id": "3368898a-f144-421a-8e53-ae5d0c6f84ca",
            "lastName": "khalid",
            "phoneNumber": "1213",
          },
          "theme": "Simplify Management",
        },
        "method": "put",
        "url": "undefined/project_service/api/projects/123",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
